pennylane/labs/resource_estimation/templates/stateprep.py

```diff
@@ -384,10 +449,128 @@ def resource_rep(cls, num_bit_flips) -> CompressedResourceOp:
         Returns:
             CompressedResourceOp: the operator in a compressed representation
         """
-
         params = {"num_bit_flips": num_bit_flips}
         return CompressedResourceOp(cls, params)
 
     @classmethod
     def tracking_name(cls, num_bit_flips) -> str:
         return f"BasisState({num_bit_flips})"
+
+
+class ResourceMPSPrep(qml.MPSPrep, ResourceOperator):
+    r"""Resource class for the MPSPrep template.
+
+    Args:
+        mps (list[TensorLike]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state
+            as a product of site matrices. See the usage details section for more information.
+        wires (Sequence[int]): wires that the template acts on. It should match the number of MPS tensors.
+        work_wires (Sequence[int]): list of extra qubits needed in the decomposition. If the maximum dimension
+            of the MPS tensors is :math:`2^k`, then :math:`k` ``work_wires`` will be needed. If no ``work_wires`` are given,
+            this operator can only be executed on the ``lightning.tensor`` device. Default is ``None``.
+        right_canonicalize (bool): indicates whether a conversion to right-canonical form should be performed to the MPS.
+            Default is ``False``.
+
+    Resource Parameters:
+        * num_wires (int): number of qubits corresponding to the state preparation register
+        * num_work_wires (int): number of additional qubits matching the bond dimension of the MPS.
+
+    Resources:
+        The resources for MPSPrep are according to the decomposition, which uses generic :class:`~.QubitUnitary`.
+        The decomposition is based on the routine described in `Fomichev et al. (2024) <https://arxiv.org/pdf/2310.18410>`_.
+
+    .. seealso:: :class:`~.MPSPrep`
+
+    **Example**
+
+    The resources for this operation are computed using:
+
+    >>> re.ResourceMPSPrep.resources(num_wires=5, num_work_wires=2)
+    defaultdict(<class 'int'>, {QubitUnitary(2): 2, QubitUnitary(3): 3})
+    """
+
+    def __init__(self, mps, wires, work_wires=None, right_canonicalize=False, id=None):
+        # Overriding the default init method to allow for CompactState as an input.
+
+        if isinstance(mps, re.CompactState):
```

Comment:
Just a fly-by comment - using a `@singledispatchmethod` might be cleaner? (not sure if it has any overheads)